From 3daa281c155da9da166627e35114283665f01669 Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <aderumier@odiso.com>
Date: Thu, 13 Jun 2019 14:34:59 +0200
Subject: [PATCH 5/7] don't remove bridge is tap|veth are still plugged

Signed-off-by: Alexandre Derumier <aderumier@odiso.com>
Signed-off-by: Thomas Lamprecht <t.lamprecht@proxmox.com>
---
 ifupdown2/ifupdown/ifupdownmain.py | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/ifupdown2/ifupdown/ifupdownmain.py b/ifupdown2/ifupdown/ifupdownmain.py
index 01163e2..b4e1864 100644
--- a/ifupdown2/ifupdown/ifupdownmain.py
+++ b/ifupdown2/ifupdown/ifupdownmain.py
@@ -2191,7 +2191,15 @@ class ifupdownMain:
                 # append it to the down list
                 newifaceobjlist = new_ifaceobjdict.get(ifname)
                 if not newifaceobjlist:
-                    ifacedownlist.append(ifname)
+                    removebridge = 0
+                    if (ifaceLinkKind.to_str(lastifaceobjlist[0].link_kind) == 'bridge'):
+                         runningbridgeports = os.listdir('/sys/class/net/%s/brif/' % ifname)
+                         for bport in runningbridgeports:
+                             if re.match('^(tap|veth|fwpr)',bport):
+                                 self.logger.warn("cant remove bridge %s, port %s is present " %(ifname,bport))
+                                 removebridge = 1
+                    if removebridge != 1:
+                        ifacedownlist.append(ifname)
                     continue
                 # If ifaceobj was present in the old interfaces file,
                 # and does not have a config in the new interfaces file
-- 
2.20.1

