From af2f2c44bea0a3a89f338d8e7cd5a8f61334a491 Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <aderumier@odiso.com>
Date: Thu, 13 Jun 2019 14:40:17 +0200
Subject: [PATCH] add uplink-id option

Signed-off-by: Alexandre Derumier <aderumier@odiso.com>
---
 ifupdown2/addons/address.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ifupdown2/addons/address.py b/ifupdown2/addons/address.py
index f576dc0..ce56ae6 100644
--- a/ifupdown2/addons/address.py
+++ b/ifupdown2/addons/address.py
@@ -111,6 +111,11 @@ class address(moduleBase):
                               'validvals': ['yes', 'no'],
                               'default' : 'no',
                               'example' : ['mpls-enable yes']},
+                      'uplink-id' :
+                            { 'help': 'uplink-id',
+                              'validrange' : ['1', '4096'],
+                              'example' : ['uplink-id 1']},
+
                     'ipv6-addrgen': {
                         'help': 'enable disable ipv6 link addrgenmode',
                         'validvals': ['on', 'off'],
-- 
2.20.1

