From cfe5feada5532830a53031138c471541ebec813d Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <aderumier@odiso.com>
Date: Sun, 17 May 2020 15:46:36 +0200
Subject: [PATCH] openvswitch : ovs-ports-condone-regex : exclude tap|veth|fwln

Signed-off-by: Alexandre Derumier <aderumier@odiso.com>
---
 ifupdown2/addons/openvswitch.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ifupdown2/addons/openvswitch.py b/ifupdown2/addons/openvswitch.py
index 468d7df..48beef3 100644
--- a/ifupdown2/addons/openvswitch.py
+++ b/ifupdown2/addons/openvswitch.py
@@ -65,6 +65,7 @@ class openvswitch(Addon, moduleBase):
             },
             'ovs-ports-condone-regex': {
                     "help": "ovs ports to ignore/condone when reloading config / removing interfaces",
+                    "default": "^(tap|veth|fwln)",
                     "required": False,
                     "example": ["ovs-ports-condone-regex ^[a-zA-Z0-9]+_v[0-9]{1,4}$"]
             },
@@ -101,6 +102,8 @@ class openvswitch(Addon, moduleBase):
 
     def _get_ovs_port_condone_regex(self, ifaceobj, get_string = False):
         ovs_port_condone_regex = ifaceobj.get_attr_value_first('ovs-ports-condone-regex')
+        if not ovs_port_condone_regex:
+            ovs_port_condone_regex = self.get_attr_default_value('ovs-ports-condone-regex')
         if ovs_port_condone_regex:
             if get_string:
                 return ovs_port_condone_regex
-- 
2.20.1

