From aaf288fd97929140348d2de6751c4ca758997455 Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <aderumier@odiso.com>
Date: Thu, 13 Feb 2020 07:16:33 +0100
Subject: [PATCH] networking.service: ifup/ifdown allow=ovs on start/stop

Signed-off-by: Alexandre Derumier <aderumier@odiso.com>
---
 debian/ifupdown2.networking.service | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/debian/ifupdown2.networking.service b/debian/ifupdown2.networking.service
index b2acd97..4347bd2 100644
--- a/debian/ifupdown2.networking.service
+++ b/debian/ifupdown2.networking.service
@@ -11,6 +11,8 @@ RemainAfterExit=yes
 SyslogIdentifier=networking
 TimeoutStopSec=30s
 ExecStart=/sbin/ifup -a
+ExecStart=/sbin/ifup --allow=ovs
+ExecStop=/sbin/ifdown --allow=ovs
 ExecStop=/sbin/ifdown -a
 ExecReload=/sbin/ifreload -a
 
-- 
2.20.1

