From ff810060d779488ba093bfad67b6cd36071ebdfd Mon Sep 17 00:00:00 2001
From: Thomas Lamprecht <t.lamprecht@proxmox.com>
Date: Tue, 25 May 2021 10:34:16 +0200
Subject: [PATCH 11/12] d/rules: add dh_installsystemd override for compat >=
 12

Signed-off-by: Thomas Lamprecht <t.lamprecht@proxmox.com>
Signed-off-by: Alexandre Derumier <aderumier@odiso.com>
---
 debian/rules | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/debian/rules b/debian/rules
index ec1b214..6539b24 100755
--- a/debian/rules
+++ b/debian/rules
@@ -24,5 +24,9 @@ override_dh_systemd_start:
 override_dh_systemd_enable:
 	dh_systemd_enable --name=networking
 
+# above are depreacated and won't be used in DH compat level >= 12
+override_dh_installsystemd:
+	dh_installsystemd --name=networking --no-start
+
 override_dh_compress:
 	dh_compress -X.py
-- 
2.30.2

